corehq/sql_db/connections.py

```diff
@@ -151,26 +170,20 @@ def _populate_connection_map(self):
                 if isinstance(db_config, dict):
                     write_db = db_config['WRITE']
                     read = db_config['READ']
-                    for db_alias, weighting in read:
-                        assert isinstance(weighting, int), 'weighting must be int'
-                        assert db_alias in settings.DATABASES, db_alias
+                    assert set(read.keys()).issubset(set(settings.DATABASES.keys()))
```

Comment:
I don't think you need `.keys()` for either of these.

Seems you've dropped the check for weighting being `int`

Reply:
Re. `keys()`, yes, I will change that. Re. weight type checking see https://github.com/dimagi/commcare-hq/pull/20315#discussion_r183657175